src/theme/Navbar/Content/index.tsx

```diff
@@ -1,23 +1,45 @@
-import React, {useState, useEffect, useContext} from "react"
+import React, {useEffect, useState} from "react"
 import {useThemeConfig, ErrorCauseBoundary} from "@docusaurus/theme-common"
 import {splitNavbarItems, useNavbarMobileSidebar} from "@docusaurus/theme-common/internal"
-import NavbarItem from "@theme/NavbarItem"
+import {useHistory} from "react-router-dom"
+import {useLocation} from "@docusaurus/router"
+import NavbarItem, {type Props as NavbarItemConfig} from "@theme/NavbarItem"
+
+import Search from "docusaurus-lunr-search/src/theme/SearchBar" // Assuming Search is a valid component
 import NavbarColorModeToggle from "@theme/Navbar/ColorModeToggle"
 import SearchBar from "@theme/SearchBar"
 import NavbarMobileSidebarToggle from "@theme/Navbar/MobileSidebar/Toggle"
 import NavbarLogo from "@theme/Navbar/Logo"
 import NavbarSearch from "@theme/Navbar/Search"
-import styles from "./styles.module.css"
 import GithubStarsButton from "@site/src/components/shared/GithubStarsButton"
+import SearchIcon from "@site/static/icons/basic/search.svg"
+import PageSearchIcon from "@site/static/icons/basic/page-search.svg"
+import styles from "./styles.module.css"
+import {getSearchInputRef, setBodyOverflow} from "@site/src/utils"
+
+type NavbarItemType =
+  | "html"
+  | "search"
+  | "default"
+  | "doc"
+  | "docsVersion"
+  | "docSidebar"
+  | "dropdown"
+  | "docsVersionDropdown"
+  | "localeDropdown"
+  | typeof NavbarItem
 
-function useNavbarItems() {
-  // TODO temporary casting until ThemeConfig type is improved
-  return useThemeConfig().navbar.items
+// Function to retrieve navbar items from the theme configuration
+const useNavbarItems = () => {
+  return useThemeConfig().navbar.items as NavbarItemConfig[]
```

Comment:
don't type cast

Reply:
This is done by docusaurus itself when we swizzle the Navbar component.